### ISSUE TITLE
Implement mechanism to register/update/delete webhook on block create/update/delete

### ISSUE THREAD

**Issue by Pwuts** [Significant-Gravitas/AutoGPT#8353]
* Part of #8352
* Blocked by #8354

## Implementation outline

* `Block.on_node_update(cls, new_default_input, old_default_input = None)` - **register/update webhook when** `credentials` **is set/updated; remove webhook when** `credentials` **is unset**
  * Defined on `Block` base class with empty (`pass`) body
  * Called with only `new_default_input` when the block is created
  * Called with `new_default_input` and `old_default_input` if `input_default` is changed
  * Implementation on webhook-triggered block should handle creating/updating the webhook
* `Block.on_node_delete(cls, default_input)` - **delete webhook when the node is deleted**
  * Defined on `Block` base class with empty (`pass`) body
* `POST /graphs/{graph_id}` endpoint calls `on_nodes_update` utility function
  * `on_nodes_update` detects which blocks / block inputs are changed on graph save and calls `Block.on_node_update` for each of them

**Comment by Pwuts** [comment]
The stuff that is necessary to make the node lifecycle hooks work is so far out of scope for the `Block` that I'm pulling it out of there and giving them their own module `backend.integrations.webhooks.node_lifecycle_hooks`.

**Comment by Pwuts** [comment]
[https://github.com/Significant-Gravitas/AutoGPT/pull/8358/commits/321ef9daae3ad0ca06c949b7a5bdce3bde226865](https://github.com/Significant-Gravitas/AutoGPT/pull/8358/commits/321ef9daae3ad0ca06c949b7a5bdce3bde226865)
